ui-v2/app/adapters/kv.js

```diff
@@ -98,7 +99,7 @@ export default Adapter.extend({
           break;
         case this.isQueryRecord(url):
           response = {
-            ...response[0],
+            ...removeNull(response[0]),
```

Comment:
What happens if/when a property has a `null` value and that should be preserved?

Reply:
Ideally if its `null` I want it to be undefined, i.e. the property shouldn't exist. I don't know if you remember but I was having weird issues with `Tags` and using `defaultValue` in the `Model` and it not setting the `defaultValue` for me. I realised it was because Tags comes in as `Tags: null` when it's empty - and then ED considers `null` to be a value.

ED also considers `null` to be a value for a 'string', which I'm not sure if I agree with and is the root of this bug.

Back to the comment - the backend has also started omitting properties that are empty from the JSON response for new API's, which I gather is preferred in go-land/backend-land also. So really this is only relevant for the older endpoints, where we'd introduce a breaking change if we removed the property.

Right now, I've stuck with my 'extremely targeted' approach to fix this, so right now I'm only removing null properties on the `queryRecord` for KV's i.e. only when I request a single KV. For my purposes right now, `Value` should never be `null` it should be undefined if its empty.

Once we get past the next release I'll be going more heavy handed and I'll probably look to fix this in a custom 'string' transformer that sets the value to '' (so an actual string) if its null/undefined.